src/test/java/com/balako/onlinebookstore/controller/CategoryControllerTest.java

```diff
@@ -0,0 +1,203 @@
+    private static final BookDtoWithoutCategoryIds DTO_WITHOUT_CATEGORY_IDS =
+            new BookDtoWithoutCategoryIds();
+    private static final String ADMIN_USERNAME = "admin";
+    private static final String ADMIN_ROLE = "ADMIN";
+    private static final String USER_USERNAME = "user";
+    private static final String SQL_SCRIPT_BEFORE_METHOD =
+            "classpath:db/book/add-book-to-book-table"
+                    + "-and-category-to-category-table.sql";
+    private static final String SQL_SCRIPT_AFTER_METHOD =
+            "classpath:db/book/delete-from-book-and-category-table.sql";
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @BeforeAll
+    static void beforeAll(@Autowired WebApplicationContext webApplicationContext) {
+        mockMvc = MockMvcBuilders.webAppContextSetup(webApplicationContext)
+                .apply(springSecurity())
+                .build();
+
+        VALID_CATEGORY_DTO_1.setId(1L);
+        VALID_CATEGORY_DTO_1.setName("Category 1");
+        VALID_CATEGORY_DTO_1.setDescription("Category 1 description");
+
+        VALID_CATEGORY_DTO_2.setId(2L);
+        VALID_CATEGORY_DTO_2.setName("Category 2");
+        VALID_CATEGORY_DTO_2.setDescription("Category 2 description");
+
+        REQUEST_DTO.setName(VALID_CATEGORY_DTO_1.getName());
+        REQUEST_DTO.setDescription(VALID_CATEGORY_DTO_1.getDescription());
+
+        DTO_WITHOUT_CATEGORY_IDS.setId(1L);
+        DTO_WITHOUT_CATEGORY_IDS.setTitle("Title 1");
+        DTO_WITHOUT_CATEGORY_IDS.setAuthor("Author 1");
+        DTO_WITHOUT_CATEGORY_IDS.setIsbn("978-0545010221");
+        DTO_WITHOUT_CATEGORY_IDS.setDescription("Some description");
+        DTO_WITHOUT_CATEGORY_IDS.setCoverImage("Image 1");
+        DTO_WITHOUT_CATEGORY_IDS.setPrice(BigDecimal.TEN);
+    }
+
+    @Test
+    @DisplayName("Test creating book dto with valid request dto")
+    @WithMockUser(username = ADMIN_USERNAME, roles = {ADMIN_ROLE})
+    @Sql(scripts = SQL_SCRIPT_AFTER_METHOD,
+            executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD)
+    void createCategory_ValidRequest_returnCategoryDto() throws Exception {
+        String jsonRequest = objectMapper.writeValueAsString(REQUEST_DTO);
+
+        MvcResult result = mockMvc.perform(post("/api/categories")
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(jsonRequest))
+                .andExpect(status().isOk())
+                .andReturn();
+        CategoryDto actual = objectMapper
+                .readValue(result.getResponse().getContentAsString(), CategoryDto.class);
+
+        assertTrue(EqualsBuilder.reflectionEquals(VALID_CATEGORY_DTO_1, actual, "id"));
+    }
+
+    @Test
+    @DisplayName("Test getting all categories dto with valid request")
+    @WithMockUser(username = USER_USERNAME)
+    @Sql(scripts = SQL_SCRIPT_BEFORE_METHOD,
+            executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD)
+    @Sql(scripts = SQL_SCRIPT_AFTER_METHOD,
```

Comment:
In my opinion, constants for script decrease readability. But IDK

Reply:
I don't think so. We can choose an understandable naming for this script, and don't show all path in every method.
Sure, naming quite bad right now, but if code will grow, I'll change naming for more preferable.

Reply:
@CodeWebWeaver I'm not sure about readability, but code duplication in this situation would be definitely a bad decision, especially when you decide to rename/move your script